enterprise/app/repo/repo.tsx

```diff
@@ -118,16 +131,57 @@ export default class RepoComponent extends React.Component<RepoComponentProps, R
     this.fetchSecrets();
   }
 
+  async loginAndLinkGithub() {
+    if (!this.props.user) {
+      await this.loginToGithub();
```

Comment:
if `loginToGithub` fails then it will continue to `linkGithubAccount`, because `loginToGithub` does `catch` on the promise. Is that OK?

Reply:
Ah, I didn't quite know how catch interacted with await. I've updated this to remove the catches and added a single try/catch to protect the whole thing.

Side note, I think async/await is provides a such nice syntax for flattening deeply nested serial promises - but the try catch syntax makes it look so ugly :(